addons/l10n_hu_edi/models/account_move.py

```diff
@@ -0,0 +1,1135 @@
+from odoo.addons.l10n_hu_edi.models.l10n_hu_edi_connection import format_bool, L10nHuEdiConnectionError
+from odoo.addons.l10n_hu_edi.models.res_company import L10N_HU_EDI_SERVER_MODE_SELECTION
+
+import base64
+import math
+from lxml import etree
+import logging
+import re
+from datetime import timedelta
+import contextlib
+from psycopg2 import OperationalError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    # === EDI Fields === #
+    l10n_hu_edi_state = fields.Selection(
+        ######################################################################################################################
+        # STATE DIAGRAM
+        # * False, rejected, cancelled --[upload]--> False, sent, send_timeout
+        # * sent --[query_status]--> sent, confirmed, confirmed_warning, rejected
+        # * confirmed, confirmed_warning --[request_cancel]--> cancel_sent, cancel_timeout
+        # * cancel_sent, cancel_pending --[query_status]--> confirmed_warning, cancel_pending, cancelled
+        # * send_timeout --[recover_timeout]--> False, send_timeout, confirmed, confirmed_warning, rejected,
+        # * cancel_timeout --[recover_timeout]--> confirmed_warning, cancel_sent, cancel_timeout, cancel_pending, cancelled
+        ######################################################################################################################
+        selection=[
+            ('sent', 'Sent, waiting for response'),
+            ('send_timeout', 'Timeout when sending'),
+            ('confirmed', 'Confirmed'),
+            ('confirmed_warning', 'Confirmed with warnings'),
+            ('rejected', 'Rejected'),
+            ('cancel_sent', 'Cancellation request sent'),
+            ('cancel_timeout', 'Timeout when requesting cancellation'),
+            ('cancel_pending', 'Cancellation request pending'),
+            ('cancelled', 'Cancelled'),
+        ],
+        string='NAV 3.0 status',
+        copy=False,
+        index='btree_not_null',
+    )
+    l10n_hu_edi_batch_upload_index = fields.Integer(
+        string='Index of invoice within a batch upload',
+        copy=False,
+    )
+    l10n_hu_edi_attachment = fields.Binary(
+        string='Invoice XML file',
+        attachment=True,
+        copy=False,
+    )
+    l10n_hu_edi_send_time = fields.Datetime(
+        string='Invoice upload time',
+        copy=False,
+    )
+    l10n_hu_edi_transaction_code = fields.Char(
+        string='Transaction Code',
+        index='trigram',
+        copy=False,
+        tracking=True,
+    )
+    l10n_hu_edi_messages = fields.Json(
+        string='Transaction messages (JSON)',
+        copy=False,
+    )
+    l10n_hu_invoice_chain_index = fields.Integer(
+        string='Invoice Chain Index',
+        help='Index in the chain of modification invoices.',
+        copy=False,
+    )
+    l10n_hu_edi_server_mode = fields.Selection(
```

Comment:
I guess technically we don't necessarily need it for querying the status of an invoice (since the transaction ID is unique).
But isn't it nice to have a banner alerting the user that the invoice was sent in test mode?